src/test/java/com/snowflake/kafka/connector/internal/TestUtils.java

```diff
@@ -369,9 +373,28 @@ public static boolean assertError(SnowflakeErrors error, Runnable func) {
     return false;
   }
 
+  /**
+   * Check Snowflake Error Code in test
+   *
+   * @param error Snowflake error
+   * @param func function throwing exception
+   * @return true is error code is correct, otherwise, false
+   */
+  public static boolean assertExceptionType(Class exceptionClass, Runnable func) {
+    try {
+      func.run();
+    } catch (Exception ex) {
+      return ex.getClass().equals(exceptionClass);
+    }
+    return false;
+  }
+
   /** @return snowflake connection for test */
   public static SnowflakeConnectionService getConnectionService() {
-    return SnowflakeConnectionServiceFactory.builder().setProperties(getConf()).build();
+    return SnowflakeConnectionServiceFactory.builder()
+        .setProperties(getConf())
+        .setTaskID("0")
```

Comment:
set taskid because topicpartition needs taskid to get the client - kind of unreliable